### PR TITLE
support Python 3.12

### PR DESCRIPTION
**Pull request by kitagry**
- support Python3.12
- unsupport Python3.8

and some changes.

- flake8 error: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
  - I change `type(hoge) == fuga` to `isinstance(hoge, fuga)`

### CODE REVIEW

**Comment by hirosassa** [comment]
I think we need re-lock to update numpy.
https://github.com/m3dev/gokart/blob/master/poetry.lock#L1224

ref: https://numpy.org/news/#numpy-1260-released

**Comment by kitagry** [comment]
Hmm. numpy 1.25 drop Python 3.8 support. https://github.com/numpy/numpy/releases/tag/v1.25.1

So, in order to we support numpy 1.26 we should drop Python 3.8.

Do you think about this? @hirosassa @Hi-king @mski-iksm @ujiuji1259

**Comment by hirosassa** [comment]
I agree with dropping Python38 coz they ended active support this August.

https://endoflife.date/python

**Comment by kitagry** [comment]
Hmm, pyarrow doesn't support python3.12 yet. https://github.com/apache/arrow/issues/37880

**Comment by hirosassa** [comment]
@kitagry hooray!! 🎉 it will be published in around 2w
https://github.com/apache/arrow/issues/37880

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
     - uses: actions/checkout@v3
     - uses: actions/setup-python@v4
       with:
-        python-version: 3.9
+        python-version: 3.11
```
Comment (kitagry) [inline .github/workflows/format.yml:19]: pyproject-flake8 doen't support python3.12. So, I use Python3.11 for format CI.

https://github.com/csachs/pyproject-flake8/issues/30

**Comment by kitagry** [comment]
@Hi-king @mski-iksm @ujiuji1259 @hirosassa Please review!

And, is it ok to remove python 3.8 test CI from required and add python 3.12 test CI to required?

**Review by Hi-king — APPROVED** [review]

**Comment by Hi-king** [comment]
@kitagry THX!

**Comment by Hi-king** [comment]
I've changed required actions :)
image

**Pull request merged**